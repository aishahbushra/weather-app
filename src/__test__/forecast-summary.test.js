import React from "react";
import { render, getByTestId } from "@testing-library/react";
import ForecastSummary from "../components/forecast-summary";

describe("ForecastSummary", () => {

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="200"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

it("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="200"
      />
    );

    expect(getByTestId("date-id")).toHaveClass("date");
    expect(getByText("mockTemperature")).toHaveClass("temperature");
    expect(getByText("mockDescription")).toHaveClass("description");
    expect(getByText("200")).toHaveClass("icon");
  });
});