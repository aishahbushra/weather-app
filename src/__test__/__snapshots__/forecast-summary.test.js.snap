// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <h2>
    <div
      class="date"
      data-testid="date-id"
    >
      <span>
        mockDate
      </span>
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      <span>
        mockTemperature
      </span>
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      <span>
        mockDescription
      </span>
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      <span>
          
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
         
      </span>
    </div>
    <button
      value="mockDate"
    >
      More Details
    </button>
  </h2>
</DocumentFragment>
`;
